magefiles/lint.go

```diff
@@ -93,6 +93,12 @@ func (Lint) Analyzers() error {
 		"-paniccheck.skip-files=_test,zz_",
 		"-zerologmarshalcheck",
 		"-zerologmarshalcheck.skip-files=_test,zz_",
+		"-protomarshalcheck",
+		// Skip generated protobuf files for this check
+		"-protomarshalcheck.skip-files=.pb",
+		// Skip our dispatch codec logic that explicitly calls MarshalVT with proto.Marshal as a fallback
+		// Skip our internal telemetry reporter which uses a prometheus proto definition that we don't control
+		"-protomarshalcheck.skip-pkg=github.com/authzed/spicedb/pkg/proto/dispatch/v1,github.com/authzed/spicedb/internal/telemetry",
```

Comment:
aren't these ignored open-ended? Can we list by file at least? or better, can we add a `// nolint` annotation like what golangci-lint does?

Reply:
Yeah, the issue is that `// nolint` needs to be implemented at the analyzer level - there isn't a way to do it at the parent framework level. There's this package that I was considering but I wasn't sure whether to treat it as in-scope for this work: https://pkg.go.dev/github.com/kyoh86/nolint#section-readme

Reply:
sounds reasonable 👍